core/src/main/java/com/graphhopper/reader/osm/OSMReader.java

```diff
@@ -346,6 +355,26 @@ protected void processWay(ReaderWay way) {
             }
         }
 
+        List<CustomArea> customAreas = estimatedCenter == null ? emptyList() : customAreaIndex.query(estimatedCenter.lat, estimatedCenter.lon);
+        // special handling for countries: since they are built-in with GraphHopper they are always fed to the
+        // encodingmanager
+        NewCountry country = null;
+        CountryRule countryRule = null;
+        for (CustomArea customArea : customAreas) {
+            Object countryCode = customArea.getProperties().get("ISO3166-1:alpha3");
+            if (countryCode != null) {
+                country = NewCountry.valueOf(countryCode.toString());
+                if (countryRule != null)
+                    LOGGER.warn("Multiple country rules found for way {}", way.getId());
+                countryRule = CountryRule.getCountryRule(country);
+            }
+        }
+        way.setTag("country", country != null ? country : NewCountry.MISSING);
+        if (countryRule != null)
+            way.setTag("country_rule", countryRule);
```

Comment:
Isn't it the other way around? In this case the way would belong to multiple countries.

I'd prefer to lookup the country in the loop and retrieve the rule afterwards:

```suggestion
        NewCountry country = NewCountry.MISSING;
        for (CustomArea customArea : customAreas) {
            Object countryCode = customArea.getProperties().get("ISO3166-1:alpha3");
            if (countryCode == null)
                continue;
            
            if (country != NewCountry.MISSING)
                LOGGER.warn("Multiple countries found for way {}", way.getId());
            
            country = NewCountry.valueOf(countryCode.toString());
        }
        way.setTag("country", country);
        
        CountryRule countryRule = CountryRule.getCountryRule(country);
        if (countryRule != null)
            way.setTag("country_rule", countryRule);
```

Reply:
Yes, fully agree.

Reply:
Logging both countrycodes would also be good.